lib/elixir/lib/gen_server.ex

```diff
@@ -226,7 +166,198 @@ defmodule GenServer do
     * http://www.erlang.org/doc/man/gen_server.html
     * http://www.erlang.org/doc/design_principles/gen_server_concepts.html
     * http://learnyousomeerlang.com/clients-and-servers
+    """
+
+  @doc """
+  Invoked when the server is started. `start_link/3` (or `start/3`) will
+  block until it returns.
+
+  `args` is the argument term (second argument) passed to `start_link/3`.
+
+  Returning `{:ok, state}` will cause `start_link/3` to return
+  `{:ok, pid}` and the process to enter its loop.
+
+  Returning `{:ok, state, timeout}` is similar to `{:ok, state}`
+  except `handle_info(:timeout, state)` will be called after `timeout`
+  milliseconds if no messages are received within the timeout.
+
+  Returning `{:ok, state, :hibernate}` is similar to
+  `{:ok, state}` except the process is hibernated before entering the loop. See
+  `handle_call/3` for more information on hibernation.
+
+  Returning `:ignore` will cause `start_link/3` to return `:ignore` and the
+  process will exit normally without entering the loop or calling `terminate/2`.
```

Comment:
Added in https://github.com/fishcakez/elixir/commit/bd62f35745e928036bcfe3ccf9731bab961e7136